### PR TITLE
Match numpy output shapes in jnp.unique for empty arrays

### PR DESCRIPTION
**Pull request by VaggelisGian**
## Description

`jnp.unique(x, axis=k)` fabricates a phantom slice when the input is empty along every dimension:

```
z = jnp.zeros((0, 0), jnp.int32)
jnp.unique(z, axis=0).shape   # (1, 0)    numpy: (0, 0)
jnp.unique(z, axis=1).shape   # (0, 1)    numpy: (0, 0)
jnp.unique(jnp.zeros((4, 0, 0)), axis=1).shape  # (4, 1, 0)  numpy: (4, 0, 0)
```

`_unique_sorted_mask` short-circuits its lexsort call with `size = 1` when `math.prod(out_shape) == 0`, because `lexsort` cannot sort zero keys of nonzero length. Collapsing existing empty slices to one matches numpy, which treats them as vacuously equal ((n, 0) inputs give (1, 0)); but the collapse also fired when no slice exists at all (`size == 0`), producing a row numpy never emits. The fix clamps to `size = min(size, 1)`: the lexsort workaround still never sees zero keys, nonempty-slice collapse is preserved, and a fully empty axis flows through `_unique`'s existing size-0 handling.

Note: value padding under concrete `size` is unchanged, but `return_index` for fully-empty inputs with a concrete size now returns shape `(0,)` instead of `(1,)`, matching numpy.

AI assistance disclosure: this fix was developed with AI coding assistance under my direction; I verified the shape matrix against NumPy, ran all tests shown below, and take responsibility for the change.

## Test Plan

New tests in `tests/lax_numpy_setops_test.py`: an unsampled deterministic test covering the discriminating fully-empty (shape, axis) pairs plus flag combos and static-size variants, and sampled shape-matrix/equal_nan cases:

```
python -m pytest tests/lax_numpy_setops_test.py -k "testUniqueEmpty" -q
  default flags: 8 failed before the fix, 15 passed after.
JAX_ENABLE_X64=1 python -m pytest tests/lax_numpy_setops_test.py -k "testUniqueEmpty" -q
  x64: 15 passed.
python -m pytest -n auto tests/lax_numpy_setops_test.py -q
  default flags: 167 passed; x64: 171 passed.
JAX_NUM_GENERATED_CASES=250 python -m pytest tests/lax_numpy_setops_test.py -k "testUniqueEmpty" -q
  189 passed - the full 184-combination shape/axis/flag matrix.
```

An 8000+-cell sweep of shapes x axes x flag combos x dtypes x memory layouts against numpy 2.5.2 shows zero divergences post-fix.

Related history in this area: #20274 / #19320 (return_inverse rank fix for NumPy 2.0).

### CODE REVIEW

File: tests/lax_numpy_setops_test.py [hunk 1]
```diff
@@ -387,6 +387,62 @@ def np_fun(x):
     jnp_fun = partial(jnp.unique, equal_nan=equal_nan)
     self._CheckAgainstNumpy(np_fun, jnp_fun, args_maker)
 
+  @jtu.sample_product(
```
Comment (jakevdp) [inline tests/lax_numpy_setops_test.py]: This seems like a *lot* of additional testing for such a small corner case.

What about deleting all the new tests and instead adding `(0, 0)` to the list of empty array shapes at the top of the file? Then you'd get coverage for this in existing tests, and also potentially uncover similar bugs in other APIs.

**Review by jakevdp — CHANGES_REQUESTED** [review]

File: tests/lax_numpy_setops_test.py (hunk 1 above)
Comment (VaggelisGian) [reply tests/lax_numpy_setops_test.py]: Done: deleted the four new tests and added (0, 0) to empty_array_shapes at the top of the file. It flows into testUnique and the isin/setdiff1d/union1d shape sweeps, which all pass against numpy in both default and x64 modes, so no other API surfaced a mismatch on this machine. 028c6a2

**Comment by jakevdp** [comment]
Thanks! Please resolve conflicts and sqaush changes into a single commit.

File: tests/lax_numpy_setops_test.py (hunk 1 above)
Comment (VaggelisGian) [reply tests/lax_numpy_setops_test.py]: Done: squashed into a single commit and rebased onto current main (one CHANGELOG conflict resolved by keeping both entries). Full setops suite passes in default and x64 modes on the rebased tree. 2d4a00d

**Review by jakevdp — APPROVED** [review]
Thanks!
